Assets/Scripts/SaveSystem/testClasses/TestPlayer.cs

```diff
@@ -0,0 +1,54 @@
+﻿using UnityEngine;
+using SaveSystem;
+
+namespace KainsTestScripts
+{
+
+    //Example of a class/component that inherits from SaveTestEntity and implements ISaveable and is saved to the SaveSystem
+    public class TestPlayer : TestEntity, ISaveableComponent
+    {
+
+        public float mana = 200;
+
+        void Start()
+        {
+            //im a normal script that someone made. look at me gooo
+
+        }
+
+
+
+
+
+
+
+        #region SAVE SYSTEM
+        [System.Serializable]
+        protected class TestPlayerSaveData : TestEntitySaveData //inherited class that is a container for data that will be saved
```

Comment:
just confirming we need to implement this for all entities?

Reply:
well, it needs to be implemented for all components that need to be saved

Reply:
i was wondering if theres a better way to do this though. maybe adding an attribute to variables you want to save can automatically handle this